vignettes/writing_vignette.Rmd

```diff
@@ -0,0 +1,200 @@
+  %\VignetteEngine{knitr::rmarkdown}
+---
+```
+
+The `<Title>` of the vignette should be meaningful. 
+
+# Markdown
+
+## Default Options
+
+    `r ''````{r setup, include=FALSE}
+    knitr::opts_chunk$set(
+      collapse = TRUE,
+      comment = "#>"
+      )
+    ```
+
+
+## Format Sections
+
+### Table of Contents
+
+Headings must be title case and start from Heading 1:
+
+```
+# Heading 1
+This is the description of my first section.
+
+## Heading 1.1
+This is the description of my first sub-section within my first section.
+
+## Heading 1.2
+This is the description of my second sub-section within my first section.
+
+```
+
+The first section gives a brief introduction of the vignette. The last sub-section 
+of the introduction should describe the packages required to run the admiral functions. 
+The admiral package should be described first.
+
+```
+# Introduction
+
+This is the introduction of my mignette.
+
+## Required Packages
+
+    `r ''````{r, warning=FALSE, message=FALSE}
+    library(admiral)
+    # <all packages used in the vignette>
+    ```
+```
+
+The `warning = FALSE` and `message = FALSE` options are there to prevent the usual messages:
+
+<span style="color: red;font-family: textmate; font-size = 12pt;" >Attaching package: 'xxxx'</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">The following objects are masked from 'package:yyyyy':</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">  fun1, fun2</span>
+
+### Conventions
+
+#### General Conventions
+
++ Any variable name, dataset name, function, argument name must be quoted with backticks: e.g. '`'ADSL'`', '`'dm'`', '`'ADT'`'
+
++ Functions must also end with `()`.
+
++ Variables name is expected to be uppercase while datasets name is expected to be lowercase.
+
++ All the codes must be executed and the output result displayed once the code is executed. Use:
+ 
+
+      `r ''````{r}
+      #<code>
+      ```
+
++ Any output created must clearly show what the function has derived. 
+It should at least show the variables/records used as input to the function and the derived 
+variables/records.
+
+#### Conventions for ADaM Workflow
+
+For vignettes describing an ADaM workflow, 
+
++ the second section will summarize the programming workflow. The first sub-section within this 
+workflow will always describe the data read to demonstrate the use of the admiral functions,
+
++ Each sub-section within the programming workflow should be tagged (e.g. [Step1] (#step)), so that 
+the user can go on the relevant section from the programming workflow (in addition to the Table of
+contents). Don’t use a tag with a number but use a meaningful name (e.g. do not use `(#link1)`,
+but use `(#this_action)`)
+
++ the last section should link to a template script. 
+
+```
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Derive/Impute End and Start Analysis Date/time and Relative Day](#datetime)
+* ...
+* [Assign `ASEQ`](#aseq)
+
```

Comment:
Added

Reply:
What do you mean by "One could also add anchor directly to the headings"?

If you do not add the anchors, the links in "Programming Workflow" will not work.

I would not use two separate code blocks but put the code block with the anchors into the code block with the sections.

Reply:
i have added an "Or" between the 2 sections.

Reply:
Sorry, it seems it was not clear what I meant. I think there should be only one code block:
````
```
# Programming Workflow

* [Read in Data](#readdata)
* [Derive/Impute End and Start Analysis Date/time and Relative Day](#datetime)
* ...
* [Assign `ASEQ`](#aseq)

## Read in Data {#readdata}
## Derive/Impute End and Start Analysis Date/time and Relative Day {#datetime}
## ...
## Assign `ASEQ` {#aseq}

# Another Section

# Example Script
ADaM | Sample Code
---- | --------------
ADxx | [ad_adxx.R](https://github.com/Roche-GSK/admiral/blob/master/inst/templates/ad_adxx.R){target="_blank"}

```
````
I wanted to show the relationship between the items in the TOC (`* [Read in Data](#readdata)`) and the subsections (`## Read in Data {#readdata}`).

Reply:
Sorry - i did not get what you wanted to say: now all clear :) If you are fine with the update, may I please ask you to approve the PR? Otherwise please let me know :)
Thanks

Reply:
After your update the "Programming Workflow" section occurs twice, at the beginning and after the example scripts. Could you correct it? I think it should look like in my previous comment.

Reply:
oops - better now?